From 54dda03ccf7d2385bffa0aa2498e0d38a1e6afed Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Wed, 1 Aug 2018 14:39:20 +0800
Subject: [PATCH 268/318] AOSCOS: drm/radeon: Modify GART TLB setting to fix
 kdump failure

After commit 0986c1a55ca64b44ee12 ("drm/radeon: stop poisoning the GART
TLB") kdump fails to boot the second kernel, so partially revert to the
old behavior.

Cc: stable@vger.kernel.org
Signed-off-by: Huacai Chen <chenhc@lemote.com>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/radeon/rs600.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/radeon/rs600.c b/drivers/gpu/drm/radeon/rs600.c
index 7cfe137d7dcd..8fb559af1a37 100644
--- a/drivers/gpu/drm/radeon/rs600.c
+++ b/drivers/gpu/drm/radeon/rs600.c
@@ -654,6 +654,9 @@ uint64_t rs600_gart_get_page_entry(uint64_t addr, uint32_t flags)
 		addr |= R600_PTE_WRITEABLE;
 	if (flags & RADEON_GART_PAGE_SNOOP)
 		addr |= R600_PTE_SNOOPED;
+#ifdef CONFIG_CRASH_DUMP
+	addr |= R600_PTE_VALID | R600_PTE_READABLE | R600_PTE_WRITEABLE;
+#endif
 	return addr;
 }
 
-- 
2.48.1

